### PR TITLE
Engine/Contest/Solvers/Weglide*: WeGlide contests

### PR DESCRIPTION
**Pull request by Scumi**
Brief summary of the changes
----------------------------
Weglide contest solvers added.

Related issues and discussions
------------------------------
loosely related: #569

### CODE REVIEW

**Comment by MaxKellermann** [comment]
Doesn't compile:
```
src/Renderer/FlightStatisticsRenderer.cpp:176:29: error: label at end of compound statement: expected statement
  case Contest::WEGLIDE_FAI:
```

The source code (or at least the commit message) should contain a link to the official WeGlide rules, so interested developers can look them up and verify your code.
Add a test to `test/src/RunOLCAnalysis.cpp` (which is misnamed these days) so developers can test this on existing IGC files without needing to run replay manually.

**Comment by MaxKellermann** [comment]
Also: rebase every time right before you push a PR branch.

**Comment by MaxKellermann** [comment]
Also: rebase every time right before you push a PR branch.

**Comment by Scumi** [comment]
> test/src/RunOLCAnalysis.cpp (which is misnamed these days)

Yes, its pretty confusing seeing real OLC contests and other decentral ones intermixed when working the code. Ideas for a naming update? Happy to provide a PR.

File: src/Engine/Contest/ContestManager.cpp [hunk 1]
```diff
@@ -39,7 +39,11 @@ ContestManager::ContestManager(const Contest _contest,
    dhv_xc_free(trace_full, true),
    dhv_xc_triangle(trace_triangle, predict_triangle, true),
    sis_at(trace_full),
-   net_coupe(trace_full)
+   net_coupe(trace_full),
+   weglide_free(),
```
Comment (MaxKellermann) [inline src/Engine/Contest/ContestManager.cpp:43]: Just a note: this kind of explicit constructor call is redundant and useless.
The other constructor calls should be done in-line where the fields are declared, but the other (existing) fields aren't initialised that way either; that's a chance code code modernisation.

**Comment by MaxKellermann** [comment]
> Yes, its pretty confusing seeing real OLC contests and other decentral ones intermixed when working the code. Ideas for a naming update? Happy to provide a PR.

Many years ago, we called the engine "Contest", but that isn't a good name either, too generic. But it's a lot better than "OLC".

Naming stuff is something I sometimes procrastinate the most with, because it's more difficult than writing code. I don't know what's the best name for this feature. Scoring? Also too generic. Not sure. Maybe standardise on "Contest" for now, just to get rid of "OLC"?

**Pull request merged**